config/metadata/basic_metadata.yaml

```diff
@@ -0,0 +1,67 @@
+attributes:
+  abstract:
+    type: string
+    multiple: true
```

Comment:
I am really liking defining metadata in a config file!!!

Have you had any thoughts on other parts of adding metadata beyond the attribute definitions...
* in WorkForm subclass... (for new/edit form)
  * terms determines which properties are shown on the new/edit form (and order I think)
  * required_fields determines which properties are required on the new/edit form
  * primary_terms determines which properties are above the fold
  * secondary_terms determines which properties are below the fold
* in WorkShowPresenter subclass... (for show page)
  * delegate solr retrieval for each term to solr_document (methods defined in solr_document.rb)
* in app/models/solr_document.rb... (for show page)
  * add method to extract value from solr_document for each property to be retrieved by the presenter
* copy in and update `app/views/curation_concerns/base/_attribute_rows.html.erb`... (for show page)
  * add `<%= presenter.attribute_to_html(:_property_) %>`  for each property
  * optionally use a `render_as` parameter to `attribute_to_html` to control how the html is written out for the property
* configure blacklight... (for discovery)

Reply:
Perhaps expand the yml to include some of this information...
```
  description:
    type: string
    multiple: true
    editable:                    # if editable not included, not editable
      fold: :primary | :secondary
      required: true | false
      index: _integer_       # sort order of fields within the primary/secondary fold)
   showable:
      render_as: :attribute | :date_attribute | :external_link_attribute | :faceted | etc. (and custom renderers)
      index: _integer_       # sort order of fields on the show page
```

The pieces this doesn't address are the solr_document extraction methods and delegators, and the blacklight config.  There may be other stuff I'm not remembering.  But this would go a long way toward making a simpler process.

Reply:
And questions in general for how these aspects are being handles in the Valkyrized Hyrax...
* controlling how a property is indexed - previously done on the property statement with `index.as`
* predicate associated with the property

Reply:
See also: https://github.com/samvera-labs/houndstooth/blob/master/examples/hyrax.yaml

My notion is that something like the `SimpleSchemaLoader` used here could be defined for other input formats: `HoundstoothSchemaLoader`, `MyCustomDatabaseSchemaLoader`, etc...

In the meanwhile, `SimpleSchemaLoader` could be extended to add support for validation specs, indexing rules, etc..., along the lines of what you're suggesting above.

Reply:
> controlling how a property is indexed - previously done on the property statement with index.as

notably, we've officially deprecated this as of today. The present way of handling this is to define a custom indexer and set it for the model. This pattern has been available since Hyrax 1.0 (i believe).

I expect to keep it, with a new implementation, for Valkyrie. If that's the case, it should be fairly easy to adopt a module builder pattern like this one for those indexers, and read the schema config (or perhaps a separate config document?) to define indexing behavior.

Reply:
Prospective implementations for indexing are now included in this PR.